includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -303,6 +305,22 @@ class AMP_Style_Sanitizer extends AMP_Base_Sanitizer {
 	 */
 	private $selector_mappings = [];
 
+	/**
+	 * The allowed viewport rules that can be extracted and moved to meta[name="viewport"].
+	 *
+	 * @var array
+	 */
+	private $allowed_viewport_rules = [ 'viewport', '-ms-viewport', '-o-viewport' ];
+
+	/**
+	 * The viewport rules that were extracted from stylesheets.
+	 *
+	 * These aren't valid in CSS, but were extracted so they can be moved to the meta[name="viewport"],
+	 *
+	 * @var array
+	 */
+	private static $extracted_viewport_rules = [];
```

Comment:
Update: instead of passing around an object, the methods should be updated to return an array that contains both the `validation_errors` _and_ the `viewport_rules` (along with `imported_stylesheet_urls`, etc). These can then be merged with the return values of any recursive calls.

Reply:
Ah, good idea. I started applying this, and will continue.